sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,232 @@
+# Authors: Andreas Mueller <andreas.mueller@columbia.edu>
+#          Guillaume Lemaitre <guillaume.lemaitre@inria.fr>
+# License: BSD 3 clause
+
+import numpy as np
+
+from ..base import BaseEstimator, RegressorMixin, is_regressor, clone
+from ..utils.validation import check_is_fitted
+from ..utils import check_X_y, safe_indexing
+from ._function_transformer import FunctionTransformer
+
+__all__ = ['TransformTargetRegressor']
+
+
+class TransformTargetRegressor(BaseEstimator, RegressorMixin):
+    """Meta-estimator to regress on a transformed target.
+
+    Useful for applying a non-linear transformation in regression
+    problems. This transformation can be given as a Transformer such as the
+    QuantileTransformer or as a function and its inverse such as ``np.log`` and
+    ``np.exp``.
+
+    The computation during ``fit`` is::
+
+        regressor.fit(X, func(y))
+
+    or::
+
+        regressor.fit(X, transformer.transform(y))
+
+    The computation during ``predict`` is::
+
+        inverse_func(regressor.predict(X))
+
+    or::
+
+        transformer.inverse_transform(regressor.predict(X))
+
+    Read more in the :ref:`User Guide <preprocessing_targets>`.
+
+    Parameters
+    ----------
+    regressor : object, (default=LinearRegression())
+        Regressor object such as derived from ``RegressorMixin``. This
+        regressor will be cloned during fitting.
+
+    transformer : object, (default=None)
+        Estimator object such as derived from ``TransformerMixin``. Cannot be
+        set at the same time as ``func`` and ``inverse_func``. If
+        ``transformer`` is ``None`` as well as ``func`` and ``inverse_func``,
+        the transformer will be an identity transformer. Note that the
+        transformer will be cloned during fitting.
+
+    func : function, optional
+
+        Function to apply to ``y`` before passing to ``fit``. Cannot be set at
+        the same time as ``transformer``. The function needs to return a
+        2-dimensional array. If ``func`` is ``None``, the function used will be
+        the identity function.
+
+    inverse_func : function, optional
+        Function to apply to the prediction of the regressor. Cannot be set at
+        the same time as ``transformer`` as well. The function needs to return
+        a 2-dimensional array. If ``inverse_func is ``None``, the function used
+        will be the identity function. The inverse function is used to return
+        to the same space of the original training labels during prediction.
```

Comment:
return to the same space ... during prediction -> return predictions to the space of the original training labels.